File: TAP_practica1/src/main/Tests/test_invoke/TestUniqueInvoke.java
package main.Tests.test_invoke;

import java.util.Map;
import java.util.function.Function;

import main.FaaS.Controller;

public class TestUniqueInvoke {

    public static void main(String[] args) {
        Controller controller = new Controller(4, 1024);
        Function<Map<String, Integer>, Integer> f = x -> x.get("x") + x.get("y");
        controller.registerAction("addAction", f, 256);

       
        try {
            int res = controller.invoke("addAction", Map.of("x", 6, "y", 2));
            res = controller.invoke("addAction", Map.of("x", 6, "y", 2));
            res = controller.invoke("addAction", Map.of("x", 6, "y", 2));
            res = controller.invoke("addAction", Map.of("x", 6, "y", 2));
            res = controller.invoke("addAction", Map.of("x", 6, "y", 2));
            System.out.println(res);
        } catch (RuntimeException e) {
            System.out.println(e.getCause().getMessage());
        }


        Function<Map<String, String>, String> f1 = x -> x.get("x") + x.get("y");
        controller.registerAction("concatAction", f1, 256);


        try {
            String res = controller.invoke("concatAction", Map.of("x", "Hola ", "y", "Mundo"));
            System.out.println(res);
        } catch (RuntimeException e) {
            System.out.println(e.getCause().getMessage());
        }
    }

}

File: TAP_practica2/src/main/scala/Tests/test_invoke/TestUniqueInvoke.scala
package Tests.test_invoke

import FaaS.Controller

import java.util.function.Function
import java.util.{HashMap => JavaHashMap, Map => JavaMap}

object TestUniqueInvoke extends App {
  val controller = new Controller(4, 1024)

  // Acción para sumar dos enteros.
  val addAction: Function[JavaMap[String, Integer], Integer] = (x: JavaMap[String, Integer]) => {
    x.getOrDefault("x", 0) + x.getOrDefault("y", 0)
  }
  controller.registerAction("addAction", addAction, 256)

  // Acción para concatenar dos cadenas.
  val concatAction: Function[JavaMap[String, String], Any] = (x: JavaMap[String, String]) => {
    x.getOrDefault("x", "") + x.getOrDefault("y", "")
  }
  controller.registerAction("concatAction", concatAction, 256)

  // Intento de invocar la acción addAction varias veces.
  try {
    var res: Any = controller.invoke("addAction", new JavaHashMap[String, Integer]() {
      {
        put("x", 6); put("y", 2)
      }
    })
    res = controller.invoke("addAction", new JavaHashMap[String, Integer]() {
      {
        put("x", 6); put("y", 2)
      }
    })
    res = controller.invoke("addAction", new JavaHashMap[String, Integer]() {
      {
        put("x", 6); put("y", 2)
      }
    })
    res = controller.invoke("addAction", new JavaHashMap[String, Integer]() {
      {
        put("x", 6); put("y", 2)
      }
    })
    res = controller.invoke("addAction", new JavaHashMap[String, Integer]() {
      {
        put("x", 6); put("y", 2)
      }
    })
    println(res.asInstanceOf[Integer])
  } catch {
    case e: RuntimeException => println(e.getCause.getMessage)
  }

  // Intento de invocar la acción concatAction.
  try {
    val res = controller.invoke("concatAction", new JavaHashMap[String, String]() {
      {
        put("x", "Hola "); put("y", "Mundo")
      }
    }).asInstanceOf[String]
    println(res)
  } catch {
    case e: RuntimeException => println(e.getMessage)
  }
}
